usagov/usagov-2021
# Usagov 1085 content lock

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1085

## Description

I enable a module that works as an extension for the module we already have. With this module we now have a configuration that allows us to configure a Lock timeout (in minutes) so that if the page lock reaches that time limit and the person locking that content is not working on the page, it will automatically break the lock. 
Module settings: http://localhost/admin/config/content/content_lock/timeout

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions


1. Run your Docker container as usual.
2. Open http://localhost/user/login in two different browsers. Example: Google Chrome and Safari.
3. Log in to two different accounts, one should be a Site Admin and the other a Content Administrator, one account on each browser.
4. On the Site Admin account, go to this page http://localhost/admin/config/content/content_lock/timeout and change the "Lock Timeout" to 1 minute, or whatever amount of time works best for you.
5. Click "Save Settings" and flush all caches on both accounts.
6. In one of the accounts, go to any page on the website and click edit to lock the content. Example: http://localhost/node/1/edit.
7. Go to the other account and try editing the same page. A message like this should appear on the screen:
<img width="1078" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/d26f7aa4-f860-4c3b-9cea-79c175854530">
8. In the same account that you used in the previous step, refresh the page as soon as the time you set in the configuration expires. A message like this should appear on the screen: 
<img width="1198" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/0235ae26-0107-4f33-8fde-628dcbee3913">
9. Try this with one or two more pages.

### Requires New Config

### Requires New Content

### Validation Steps
- Content Administrator, or any other account with permissions, should be able to edit a page if it has not been edited for the amount of time specified in the settings. The time that will be used now is 24 hours (1440 minutes).

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
11. Find the commit of this pull request.
12. Build and deploy the changes.
13. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
